nautobot/extras/models/mixins.py

```diff
@@ -18,3 +25,51 @@ def dynamic_groups(self):
             self._dynamic_group_queryset = queryset
 
         return self._dynamic_group_queryset
+
+
+# 2.0 TODO: Remove after v2 since we will no longer care about backwards-incompatibility.
+class TaggableManagerMonkeyMixin:
+    """
+    Dynamically-applied monkey-patch mixin that is used to replace any defined
+    methods on eligible objects.
+
+    Intended to be used on the right-hand side of `monkey_mix()` to overload the
+    `formfield()` method.
+
+    Usage:
+
+        monkey_mix(TaggableManager, TaggableManagerMonkeyMixin)
+
+    See: `nautobot.extras.apps.ready()`
+    """
+
+    @once_per("cls")
+    def _install_hotfix(self, cls):
+        # Install auto-created models as their module attributes to make them picklable
+        module = sys.modules[cls.__module__]
+        if not hasattr(module, cls.__name__):
+            setattr(module, cls.__name__, cls)
+
+    # This is probably still needed if models are created dynamically.
+    def contribute_to_class(self, cls, name):
```

Comment:
```suggestion
    def contribute_to_class(self, cls, name):
        """
        Overload default so that we can assert that this is called when 
        attached to any model that is using a `TaggableManager`.
        
        Using `.contribute_to_class()` is how field objects get added to the model
        at during the instance preparation. This is also where any custom model
        methods are hooked in. So in short this method asserts that any time a
        `TaggableManager` is added to a model, that model also gets its methods
        monkey-mixed without having to define them on the model yourself.
        """    
```